File: programmingclub/teleport/teleport.java
import java.io.*;
import java.util.*;

public class teleport {
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(new File("teleport.in"));
        PrintWriter out = new PrintWriter("teleport.out");

        int a = in.nextInt();
        int b = in.nextInt();
        int x = in.nextInt();
        int y = in.nextInt();

        int d1 = Math.abs(a - b);
        int d2 = Math.abs(a - x) + Math.abs(b - y);
        int d3 = Math.abs(a - y) + Math.abs(b - x);

        out.println(Math.min(d1, Math.min(d2, d3)));

        out.close();
    }
}

File: programmingclub/teleport/teleport.py
inp = open('teleport.in', 'r')
out = open('teleport.out', 'w')

A, B, X, Y = [int(i) for i in inp.readline().split()]

d1 = abs(A - B)
d2 = abs(A - X) + abs(B - Y)
d3 = abs(A - Y) + abs(B - X)
min_d = min(d1, min(d2, d3))

out.write(str(min_d))
